### PR TITLE
Update readme to standard format

### PR DESCRIPTION
**Pull request by gilbertw**

### CODE REVIEW

File: sdk/core/core/README.md [hunk 1]
```diff
@@ -1,10 +1,16 @@
-# Azure SDK Core Library
+# Azure SDK Core Library for C
 
-Azure Core library (`az_core`) provides shared primitives, abstractions, and helpers for modern Azure SDK client libraries written in the C programming language. These libraries follow the Azure SDK Design Guidelines for Embedded C.
+Azure Core library for C (`az_core`) provides shared primitives, abstractions, and helpers for modern Azure SDK client libraries written in the C programming language. These libraries follow the Azure SDK Design Guidelines for Embedded C.
```
Comment (JeffreyRichter) [inline sdk/core/core/README.md]: Do we use the 'az_core' anywhere else? It is not in the code base at all. Perhaps remove this parenthetical?

File: sdk/core/core/README.md [hunk 2]
```diff
@@ -34,7 +40,7 @@ inline bool az_succeeded(az_result result) {
 
 Additional information could be passed using output parameters.
 
-## 2. Span of bytes
+### Span of bytes
 
 Defined in [inc/az_span.h](inc/az_span.h).
```
Comment (JeffreyRichter) [inline sdk/core/core/README.md]: These code snippets are out of date now.

File: sdk/core/core/README.md [hunk 3]
```diff
@@ -50,10 +56,81 @@ typedef struct {
 } az_span;
 ```
 
-## 3. Strings
+### Strings
 
 A string is a span of UTF-8 characters. It's not a zero-terminated string. Defined in [inc/az_str.h](inc/az_str.h).
```
Comment (JeffreyRichter) [inline sdk/core/core/README.md]: This header no longer exists.

File: sdk/core/core/README.md [hunk 4]
```diff
@@ -50,10 +56,81 @@ typedef struct {
...
+
+### More sample code
+
+TODO
+
+### Additional documentation
+
+TODO
+
+## Contributing
```
Comment (JeffreyRichter) [inline sdk/core/core/README.md]: I assume that this is all boilerplate and identical to our other SDK repos. I have no opinion on this part of the readme.

File: sdk/identity/identity/README.md [hunk 5]
```diff
@@ -1,3 +1,99 @@
-# Azure SDK Identity Library
+# Azure SDK Identity Library for C
```
Comment (JeffreyRichter) [inline sdk/identity/identity/README.md]: We're going to get rid of this directory and put "credentials" (which used to be called identity) in the core directory.

File: sdk/identity/identity/README.md [hunk 6]
```diff
@@ -1,3 +1,99 @@
...
 
 Azure Identity library (`az_identity`) provides primitives, abstractions, and helpers for Azure authorization.
+
+## Getting started
+
+### Install the package
+
+TODO link to the vcpkg
+
+**Prerequisites**: You must have an [Azure subscription][azure_sub] and [C++ compiler][cpp_compiler] to use this package.
```
Comment (JeffreyRichter) [inline sdk/identity/identity/README.md]: This says C++ compiler; but C required.

File: sdk/keyvault/keyvault/README.md [hunk 7]
```diff
@@ -1,7 +1,114 @@
-# Azure SDK KeyVault Library
+# Azure SDK KeyVault Library for C
 
 Azure KeyVault library (`az_keyvault`) provides abstractions, and helpers for communicating with Azure KeyVault in the C programming language. This library follows the Azure SDK Design Guidelines for Embedded C.
```
Comment (JeffreyRichter) [inline sdk/keyvault/keyvault/README.md]: It's actually a KeyVault Keys (az_keyvault_keys) library as we do not support other features of keyvault. However, I thought we were going to make this a sample and not a formal library that we support externally.

File: sdk/keyvault/keyvault/README.md [hunk 8]
```diff
@@ -1,7 +1,114 @@
...
 
-The library allows clients to communicate with Azure KeyVault.
+Use the library to communicate with Azure KeyVault.
```
Comment (JeffreyRichter) [inline sdk/keyvault/keyvault/README.md]: Azure KeyVault Keys

File: sdk/storage/blobs/README.md [hunk 9]
```diff
@@ -1,7 +1,114 @@
-# Azure SDK Storage Blobs Library
+# Azure SDK Storage Blobs Library for C
 
 Azure Storage Blobs library (`az_storage_blobs`) provides abstractions, and helpers for communicating with Azure Storage Blobs in the C programming language. This library follows the Azure SDK Design Guidelines for Embedded C.
 
-The library allows clients to communicate with Azure Storage Blobs.
+Use the Azure SDK Storage Blobs library for C to communicate with Azure Storage Blobs.
```
Comment (JeffreyRichter) [inline sdk/storage/blobs/README.md]: I wouldn't use the word "communicate"; I'd instead say: Use the Azure SDK Storage Blobs library for C to work with Azure Storage Blobs.

File: sdk/storage/blobs/README.md [hunk 10]
```diff
@@ -1,7 +1,114 @@
...
+* Upload blobs
+* Create and modify blob metadata
+
+## Getting started
+
+### Install the package
+
+TODO link to the vcpkg
+
+**Prerequisites**: You must have an [Azure subscription][azure_sub], [Azure Storage account][storage_account] (Storage Blobs), and [C++ compiler][cpp_compiler] to use this package.
```
Comment (JeffreyRichter) [inline sdk/storage/blobs/README.md]: C compiler

Comment (JeffreyRichter) [reply sdk/storage/blobs/README.md]: Technically speaking, the user of this library does NOT need to have an azure subscription nor do they need to have a storage account. Data can be uploaded to a BLOB via a SAS URL to anyone's subscription and anyone's storage account as long as they have a SAS URL --> this is the IoT scenario.

File: sdk/core/core/README.md (hunk 2 above)
Comment (gilbertw) [reply sdk/core/core/README.md]: Once that is checked in we can cleanup these readmes.  This checkin makes them follow the standard format but does not address all inaccuracies.

File: sdk/core/core/README.md (hunk 4 above)
Comment (gilbertw) [reply sdk/core/core/README.md]: Correct

File: sdk/storage/blobs/README.md (hunk 10 above)
Comment (gilbertw) [reply sdk/storage/blobs/README.md]: Removed that section

File: sdk/identity/identity/README.md (hunk 6 above)
Comment (gilbertw) [reply sdk/identity/identity/README.md]: Switched to c and added link to c compiler

File: sdk/core/core/README.md (hunk 1 above)
Comment (gilbertw) [reply sdk/core/core/README.md]: In CMake files we use it to reference the package.  Its relevant for those building packages on top of core

File: sdk/core/core/README.md (hunk 3 above)
Comment (gilbertw) [reply sdk/core/core/README.md]: Fixed, also fixed the structure for az_span

File: sdk/identity/identity/README.md (hunk 5 above)
Comment (gilbertw) [reply sdk/identity/identity/README.md]: Is the package going away as well?  Are all languages doing this?

File: sdk/keyvault/keyvault/README.md (hunk 7 above)
Comment (gilbertw) [reply sdk/keyvault/keyvault/README.md]: The readme can be updated library->sample when we move the package

File: sdk/identity/identity/README.md (hunk 5 above)
Comment (gilbertw) [reply sdk/identity/identity/README.md]: This readme to be removed when the Identity package is folded into core

**Pull request merged**